File: fleet/controllers/vehicle_log.py
import frappe

@frappe.whitelist()
def create_journal_entry(source_name):
    vehicle_log = frappe.get_doc("Vehicle Log",source_name)
    journal_entry = frappe.new_doc("Journal Entry")
    journal_entry.vehicle_log = vehicle_log.name
    return journal_entry

@frappe.whitelist()
def create_purchase_invoice(source_name):
    vehicle_log = frappe.get_doc("Vehicle Log",source_name)
    purchase_invoice = frappe.new_doc("Purchase Invoice")
    purchase_invoice.vehicle_log = vehicle_log.name
    return purchase_invoice

File: fleet/public/js/vehicle_log.js
// Copyright (c) 2023, DTS and contributors
// For license information, please see license.txt

frappe.ui.form.on("Vehicle Log", {
	refresh(frm) {
		if(frm.doc.docstatus == 1) {
			frm.remove_custom_button(('Expense Claim'),('Create'));
			frm.add_custom_button(__('Journal Entry'), function() {
				frappe.model.open_mapped_doc({	
					method: "fleet.controllers.vehicle_log.create_journal_entry",
					frm: frm,
				});
			}, __('Create'));
            frm.add_custom_button(__('Purchase Invoice'), function() {
				frappe.model.open_mapped_doc({	
					method: "fleet.controllers.vehicle_log.create_purchase_invoice",
					frm: frm,
				});
			}, __('Create'));
			frm.page.set_inner_btn_group_as_primary(__('Create'));
		}
	},
});